core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -186,7 +186,7 @@ public static Catalog loadCatalog(
           String.format("Cannot initialize Catalog, %s does not implement Catalog.", impl), e);
     }
 
-    if (catalog instanceof Configurable) {
+    if (hadoopConf != null && catalog instanceof Configurable) {
```

Comment:
I wonder if we can create a separate method for the same here but without `Configuration`/`Configurable` as a parameter referenced at all, so that we can create GlueCatalog without referring to any class within the hadoop library at all